src/pages/RepoPage/CoverageTab/subroute/FileExplorer/FlagMultiSelect.jsx

```diff
@@ -0,0 +1,98 @@
+import eq from 'lodash/eq'
+import isUndefined from 'lodash/isUndefined'
+import PropTypes from 'prop-types'
+import { useState } from 'react'
+
+import { useLocationParams } from 'services/navigation'
+import { useRepoBackfilled, useRepoFlagsSelect } from 'services/repo'
+import { useFlags } from 'shared/featureFlags'
+import Icon from 'ui/Icon'
+import MultiSelect from 'ui/MultiSelect'
+
+const defaultQueryParams = {
+  // without this search here, when updating the params it fills it in for some reason :thinking:
+  search: '',
+  flags: [],
+}
+
+function FlagMultiSelect() {
+  const { params, updateParams } = useLocationParams(defaultQueryParams)
+  const [selectedFlags, setSelectedFlags] = useState(params?.flags)
+  const [flagSearch, setFlagSearch] = useState(null)
```

Comment:
Shouldn't we initialize this as `[]`?

Reply:
It would be if no flags are selected. If there are we want those flags to be set as selected when the user refreshes the page so that the multi-select stays in sync with the selected items.